pkg/userdata/centos/provider.go

```diff
@@ -277,6 +290,19 @@ write_files:
   content: |
 {{ containerRuntimeHealthCheckSystemdUnit | indent 4 }}
 
+{{- if .NodeImageRegistry }}
+- path: /run/containers/registries.conf
+  permissions: "0644"
+  content: |
+    INSECURE_REGISTRY="--insecure-registry {{ .NodeImageRegistry }}"
+{{- end}}
+
+- path: /etc/systemd/system/docker.service.d/environment.conf
```

Comment:
Can we avoid rendering this when no proxy is set?

Reply:
You can set `EnvironmentFile=-/path/to/file`, then systemd will start the unit even when the env file is absent

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/559/commits/afd9104ec6e27674aaf8885e1a8e107b8c309e48